sherpa/plot/pylab_backend.py

```diff
@@ -865,3 +862,43 @@ def plotfunc():
 as_html_fitcontour = as_html_contour
 as_html_contour1d = as_html_plot
 as_html_contour2d = as_html_contour
+
+# Needed for datastack plotting wrapper
+
+
+def initialize_plot(dataset, ids):
+    """Create the plot window or figure for the given dataset.
+
+    Parameters
+    ----------
+    dataset : str or int
+       The dataset.
+    ids : array_like
+       The identifier array from the DataStack object.
+
+    See Also
+    --------
+    select_plot
+
+    """
+    plt.figure(ids.index(dataset['id']) + 1)
```

Comment:
I'm a bit concerned with this (in that I can finally see it's been moved from the datastack code to here). The parameter description says `dataset` is a `str` or `int` but here you're accessing it by `id`, some something is wrong.

As we don't have a test of this we can't tell.

Reply:
I think this is related to #1188  - the datastack plots that are supposed to summarize the stack in some way (not clear how - no docs, no example plot and does not work in main) are all broken right now. if at all possible, I don't want to roll that into the current PR though. They were broken (and obviously not tested) before, and they are broken now. This code is just copied over. I do think that datastack plotting needs a closer look and I do think that datastack is underused. However, I also strongly feel that this is not the PR to do that; here, I just want to make it easier to switch the backends and to do that I need to merge the datastack plotting code into this module to reduce the number of places where the plotting backend is defined independently.

 #1188 is open to remind us of the problem for now. In fact, I'm doing two science projects with datastacks now, so I hope that I get some ideas there who type of plots we actually want.